### ISSUE TITLE
Fix keyboard navigation in embedded DartPad

### ISSUE THREAD

**Issue by atsansone** [dart-lang/dart-pad#3033]
Hotlist: https://buganizer.corp.google.com/hotlists/5891206
Bug: https://buganizer.corp.google.com/issues/338110557

### Observed Results

The navigation using keyboard get stuck after entering the first box for executing commands.

* [Screencast Voiceover](http://screencast/cast/NTk1NDQzNTgzMzM5NzI0OHxjZDIxZWVkZi02ZQ)
* [Screencast ChromeVox](http://screencast/cast/NTU3NjU3MTU2NDg1MTIwMHw2YWNhYjJjMC03NA)

### Expected Results

Every link/button should be accessible using the keyboard, the screen reader focus must reach all the controls in the page.

### User Impact

Users with cognitive issue or that cannot rely in the use of mouse and use keyboard instead, will find a challenging time since the navigation cannot be continued, and they will not be able to see the rest of the page..

### Steps To Reproduce

1. Enabled VoiceOver using <kbd>Cmd</kbd> + <kbd>F5</kbd> or ChromeVox using <kbd>Ctrl</kbd> + <kbd>Alt</kbd> + <kbd>Z</kbd>.
1. Open https://dart.dev/codelabs/async-await
1. Start the linear navigation using <kbd>Tab</kbd> key.
1. Observe that when the focus reaches the first box for ¨Incorrectly using an asynchronous function¨, it is trapped and the navigation with keyboard cannot be continued.

### Relevent GAR Sheet

[A11Y Tracker](https://a11ytracker.corp.google.com/products/034ba691-ac78-4cc2-88c2-e06449cfd17e/edit/testresults?useCase=3kvph5kjcvnng&e=ProductLabels).

### GAR Details

[2.1 Is the primary method of navigating and reaching content and controls accessible?](go/GAR-web#accessible-navigation), [1.3. Is focus always set correctly?](http://go/gar-web-focus), [1.2. Can all controls be reached and used via the keyboard without using a mouse?](http://go/gar-web-keyboard).

* Machine: macOS/ChromeBook
* Browser: Chrome Version 120.0.6099.272 (Official Build) (64-bit)
* Screen reader: VoiceOver/ChromeVox.
* Bug filling time: 15 mins.

**Comment by RedBrogdon** [comment]
Transferred this on behalf of @atsansone.
